src/chocolatey/infrastructure.app/services/PackageScriptService.cs

```diff
@@ -0,0 +1,79 @@
+namespace chocolatey.infrastructure.app.services
+{
+    using System.Collections.Generic;
+    using System.IO;
+    using System.Linq;
+    using domain;
+    using filesystem;
+    using results;
+
+    public class ChocolateyScriptDetail
+    {
+        public ChocolateyScriptDetail(string scriptName, string scriptPath, bool ignoreScriptErrors = false)
+        {
+            Name = scriptName;
+            Path = scriptPath;
+            IgnoreScriptErrors = ignoreScriptErrors;
+        }
+
+        public string Name { get; private set; }
+
+        public string Path { get; private set; }
+
+        public bool IgnoreScriptErrors { get; private set; }
+    }
+
+    public class PackageScriptSet : List<ChocolateyScriptDetail>
+    {
+        public PackageScriptSet(string packageInstallLocation)
+        {
+            PackageDirectory = packageInstallLocation;
+        }
+
+        public string PackageDirectory { get; private set; }
+    }
+
+    public class PackageScriptService
+    {
+        private readonly IFileSystem _fileSystem;
+
+        public PackageScriptService(IFileSystem fileSystem)
+        {
+            _fileSystem = fileSystem;
+        }
+
+        private void append_script_if_present(string packageDirectory, string scriptName, PackageScriptSet appendTo)
+        {
+            var scriptPath = _fileSystem.get_files(packageDirectory, scriptName, SearchOption.AllDirectories).FirstOrDefault();
+            if (scriptPath != null) appendTo.Add(new ChocolateyScriptDetail(scriptName, scriptPath));
+        }
+
+        public PackageScriptSet get_scripts_to_run(PackageResult packageResult, CommandNameType command)
+        {
+            var packageDirectory = packageResult.InstallLocation;
+            var scripts = new PackageScriptSet(packageDirectory);
+
+            this.Log().Debug("Locating scripts for command {0} in location {1}", command, packageDirectory);
+
+            switch (command)
+            {
+                case CommandNameType.install:
+                    append_script_if_present(packageDirectory, "chocolateyInstall.ps1", scripts);
+                    break;
+
+                case CommandNameType.upgrade:
+                    append_script_if_present(packageDirectory, "chocolateyBeforeModify.ps1", scripts);
```

Comment:
I should perhaps also clarify... run_package_scripts is intended to run one (or more, in the case of uninstall) scripts against a single package & version (i.e. in a single folder).